code/modules/mob/living/silicon/robot/robot_modules.dm

```diff
@@ -50,6 +50,8 @@
 		R.remove_module() //Helps remove screen references on robot end
 
 	for(var/obj/A in modules)
+		if(istype(A, /obj/item/weapon/storage))
+			A:empty_contents_to(loc)
```

Comment:
make sure this won't end with the borg trying to drop crap inside nullspace when the singulo eats it